cmake/BLTOptions.cmake

```diff
@@ -66,6 +66,7 @@ set(BLT_CLANG_HIP_ARCH "gfx906" CACHE STRING "AMDGPU ISA target to use when gene
 mark_as_advanced(BLT_CLANG_HIP_ARCH)
 option(ENABLE_HCC         "Enable HCC support" OFF)
 set(BLT_ROCM_ARCH "gfx900" CACHE STRING "gfx architecture to use when generating ROCm code")
+set(CMAKE_HIP_ARCHITECTURES "${BLT_ROCM_ARCH}" CACHE STRING "")
```

Comment:
Can you move this to https://github.com/LLNL/blt/blob/develop/cmake/thirdparty/SetupHIP.cmake

This file is for command line options and the rest of the HIP setup logic is in that file.

Thanks!

Reply:
Possibly, that's a good policy question.  If only the BLT one is set it's pretty clear what to do, if the CMAKE one is set, probably pull it out into the BLT one, what if they are both set?  I would think either we combine them, since it's a list, or error out.  Do you have a preference?

Reply:
I agree that if one or the other is set to just set them both to the same thing.

If both are set to different values, I would lean towards erroring out.  Thoughts, @kennyweiss ?

Reply:
I agree -- if there are two conflicting config arguments we should error out with a useful error message.

Are these options really equivalent? If so, do we still need both? 
e.g. is it:
* for backwards compatibility with different versions of CMake?
* for symmetry with how we handle other architectures?
* <something else>?

Reply:
Yeah, it's backwards compatibility.  I could always go in and replace the blt option with CMAKE_HIP_ARCHITECTURES inside of BLT for all versions, it would remove the duplication and make usage a bit more consistent across versions, but it would be an interface break for blt users.

Reply:
Come to find out, the BLT_ROCM_ARCH option wasn't used anywhere.  As in it was literally never referenced elsewhere in the repo, so I'm removing it and replacing it with the CMAKE standard option from 3.21+ and adding a reference to it so it gets propagated correctly in older versions.

Doing that, the behavior is identical between `ENABLE_HIP` and `ENABLE_CLANG_HIP` for this flag, hip takes the same flag as clang does to propagate this option, which makes `CMAKE_HIP_ARCHITECTURES` effectively a duplicate of `BLT_CLANG_HIP_ARCH`.
I'm thinking my preference is this:

* Allow users to specify either `-DCMAKE_HIP_ARCHITECTURES` or the `HCC_AMDGPU_TARGET` environment variable to tell cmake what to do
* Remove `BLT_CLANG_HIP_ARCH`, use the value from first bullet for this case

Honestly it would be nice to do the same thing for the cuda support and merge out the `BLT_CLANG_CUDA_ARCH` as well.

Reply:
Thanks for finding that out @trws !  As far as I know @davidbeckingsale is the only one that used the clang cuda support.  I do like having both options the act the same way and as close to what the CMake standard is.